### ISSUE TITLE
`App::default_service` with a guarded `Route` is confusing

### ISSUE THREAD

**Issue by mkantor** [actix/actix-web#2598]
## Expected Behavior

Consider this `App`:
```rs
App::new().default_service(web::get().to(HttpResponse::NoContent))
```

That reads like it would respond to `GET /any/path` with `HTTP 204` (which it does). However it doesn't obviously imply (to me, anyway) that the app will now also respond to, say, `CONNECT` requests with `HTTP 204`.

Weirder code like this is also unintuitive:
```rs
App::new().default_service(web::route().guard(guard::fn_guard(|_| false)).to(HttpResponse::Ok))
```

## Current Behavior

`Route`s passed to `App::default_service` have their guards ignored.

## Possible Solutions

I think ideally this sort of code would not even compile, but there doesn't appear to be a type-level distinction between guarded and un-guarded `Route`s at the moment. Alternatively, `default_service` could actually apply guards, but that may somewhat defeat its purpose.

It would at least be helpful to have a note in [the docblock for `App::default_service`](https://github.com/actix/actix-web/blob/008753f07a30bd5dbc875545f5012075ad88ae0d/src/app.rs#L239-L268) explaining that guards are ignored.

## Context

I was passing `Route` created via `web::get().to(_)` to `App::default_service` and did not realize that I had accidentally configured my server to respond to all HTTP methods.

## Your Environment

- Rust Version (I.e, output of `rustc -V`): `1.56.1`
- Actix Web Version: `3.3.3`, `4.0.0-beta.21`

**Comment by mkantor** [comment]
I wrote a couple app.rs tests to confirm the behavior before filing this issue. In case they're helpful, here they are:

```rs
#[actix_rt::test]
async fn test_default_service_is_guarded_1() {
    let srv =
        init_service(App::new().default_service(web::get().to(HttpResponse::NoContent))).await;

    let req = TestRequest::with_uri("/blah")
        .method(Method::TRACE)
        .to_request();
    let resp = srv.call(req).await.unwrap();
    assert_eq!(resp.status(), StatusCode::NOT_FOUND);
}

#[actix_rt::test]
async fn test_default_service_is_guarded_2() {
    let srv = init_service(
        App::new().default_service(
            web::route()
                .guard(guard::fn_guard(|_| false))
                .to(HttpResponse::Ok),
        ),
    )
    .await;

    let req = TestRequest::with_uri("/blah")
        .method(Method::GET)
        .to_request();
    let resp = srv.call(req).await.unwrap();
    assert_eq!(resp.status(), StatusCode::NOT_FOUND);
}
```

Those both fail in 008753f07a30bd5dbc875545f5012075ad88ae0d.

**Comment by robjtede** [comment]
fixed in #2614
